Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1092,13 +1092,59 @@ namespace Babylon
     {
         const auto textureDestination = info[0].As<Napi::Pointer<Graphics::Texture>>().Get();
         const auto textureSource = info[1].As<Napi::Pointer<Graphics::Texture>>().Get();
+        const auto flipY = info.Length() <= 2 ? true : info[2].As<Napi::Boolean>().Value();
 
-        arcana::make_task(m_update.Scheduler(), *m_cancellationSource, [this, textureDestination, textureSource, cancellationSource = m_cancellationSource]()
+        Graphics::DeviceContext& context = Graphics::DeviceContext::GetFromJavaScript(info.Env());
+
+        arcana::make_task(m_update.Scheduler(), *m_cancellationSource, [&context, this, flipY, textureDestination, textureSource, cancellationSource = m_cancellationSource]()
         {
-            return arcana::make_task(m_runtimeScheduler, *m_cancellationSource, [this, textureDestination, textureSource, updateToken = m_update.GetUpdateToken(), cancellationSource = m_cancellationSource]()
+            return arcana::make_task(m_runtimeScheduler, *m_cancellationSource, [&context, this, flipY, textureDestination, textureSource, updateToken = m_update.GetUpdateToken(), cancellationSource = m_cancellationSource]()
             {
                 bgfx::Encoder* encoder = m_update.GetUpdateToken().GetEncoder();
-                GetBoundFrameBuffer(*encoder).Blit(*encoder, textureDestination->Handle(), 0, 0, textureSource->Handle());
+
+                // Flip texture in memory (maybe there is a way for bgfx to allow flipping to happen on the GPU?)
```

Comment:
Going to CPU seems like it's unnecessary. At worst case, we should be able to flip using a shader?